### ISSUE TITLE
1.2.0 update breaks Windows

### ISSUE THREAD

**Issue by supercow** [puppetlabs/r10k#100]
Prior to 1.2.0, r10k could run on Windows machines. New changes in process management in 1.2.0 have broken this. Now, when running r10k on Windows, users will see errors about an undefined `Fnctl` constant. For example:

```
C:\Users\me\src> r10k -v -t puppetfile install
[R10K::Task::Puppetfile::Sync - INFO] Loading modules from Puppetfile into queue
[R10K::Task::Module::Sync - INFO] Deploying connection-validator into C:/Users/me/src/modules
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Module::Sync:0x2e6c7a0> failed while running: uninitialized constant Fcntl
::F_SETFD
[R10K::Task::Module::Sync - INFO] Deploying pkgdownload into C:/Users/me/src/modules
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Module::Sync:0x2e6c7d0> failed while running: uninitialized constant Fcntl
::F_SETFD
[R10K::Task::Module::Sync - INFO] Deploying staging into C:/Users/me/src/modules
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Module::Sync:0x2e6c800> failed while running: uninitialized constant Fcntl
::F_SETFD
[R10K::Task::Module::Sync - INFO] Deploying stdlib into C:/Users/me/src/modules
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Module::Sync:0x2e6c830> failed while running: uninitialized constant Fcntl
::F_SETFD
```

Regression here: d1400f387

**Comment by vernetto** [comment]
same problem here :o)

**Comment by adrienthebo** [comment]
What version of Ruby is being used with r10k on Windows in this instance? This is very easy to solve with Ruby 1.9, and utterly nightmarish on Ruby 1.8.

**Comment by supercow** [comment]
I'm personally running this with a newer PE's ruby, so 1.9. I don't mean to speak for everyone, but I think 1.9 support would probably be sufficient if 1.8 is too much effort.

**Comment by vernetto** [comment]
ruby 1.9.3p448 (2013-06-27) [i386-mingw32]  here

**Comment by leon-strong** [comment]
Yeah, i'm seeing the same issue, running 1.9 from the cygwin version of ruby, downgraded and voila, works a treat. Looks like something hinky with the new process management module.

**Comment by adrienthebo** [comment]
I've implemented Windows support for commands in https://github.com/adrienthebo/r10k/tree/defect/1.2.x/gh-100-windows-support ; please clone that branch and see if it solves your issues.

**Comment by jasonjclifford** [comment]
This works for me now. Thanks

**Comment by adrienthebo** [comment]
Excellent! I can cut a release for this very soon, but before I continue your main use case for this is for `r10k puppetfile install`, correct?

**Comment by jasonjclifford** [comment]
Yes, that is correct.

**Comment by adrienthebo** [comment]
Merged into master in 126830a and released in 1.2.1. I'm closing this issue; if this is still a problem please reopen this issue. Thanks for the issue report!
